x/pricefeed/simulation/genesis.go

```
@@ -2,21 +2,96 @@ package simulation
+
+}
+
+// getPricefeedSimulationParams returns the params with xrp:usd, btc:usd, bnb:usd
+func getPricefeedSimulationParams() types.Params {
+	// SET UP THE PRICEFEED GENESIS STATE
+	pricefeedGenesis := pricefeed.GenesisState{
+		Params: pricefeed.Params{
+			Markets: []pricefeed.Market{
+				pricefeed.Market{MarketID: "btc:usd", BaseAsset: "btc", QuoteAsset: "usd", Oracles: []sdk.AccAddress{}, Active: true},
+				pricefeed.Market{MarketID: "xrp:usd", BaseAsset: "xrp", QuoteAsset: "usd", Oracles: []sdk.AccAddress{}, Active: true},
+				pricefeed.Market{MarketID: "bnb:usd", BaseAsset: "bnb", QuoteAsset: "usd", Oracles: []sdk.AccAddress{}, Active: true},
+			},
+		},
+
+		// TODO QUESTION - are these prices necessary? they are re-set by getInitialPrice. assuming the below can be deleted?
+		PostedPrices: []pricefeed.PostedPrice{
+			// Bitcoin
+			pricefeed.PostedPrice{
+				MarketID:      "btc:usd",
+				OracleAddress: sdk.AccAddress{},
+				Price:         sdk.MustNewDecFromStr("7000.00"),
+				Expiry:        time.Now().Add(1 * time.Hour),
+			},
+			// Binance coin
+			pricefeed.PostedPrice{
+				MarketID:      "bnb:usd",
+				OracleAddress: sdk.AccAddress{},
+				Price:         sdk.MustNewDecFromStr("14.00"),
+				Expiry:        time.Now().Add(1 * time.Hour),
+			},
+			// XRP ripple coin
+			pricefeed.PostedPrice{
+				MarketID:      "xrp:usd",
+				OracleAddress: sdk.AccAddress{},
+				Price:         sdk.MustNewDecFromStr("0.2"),
+				Expiry:        time.Now().Add(1 * time.Hour),
+			},
+		},
+	}
+	return pricefeedGenesis.Params
+}
+
+func getInitialPrice(marketId string) (price sdk.Dec) {
+	switch marketId {
+	case "btc:usd":
+		return sdk.MustNewDecFromStr("7000") // TODO QUESTION - add some randomization?
+	case "bnb:usd":
+		return sdk.MustNewDecFromStr("14")
+	case "xrp:usd":
+		return sdk.MustNewDecFromStr("0.2")
+	}
+
+	fmt.Printf("Invalid marketId in getInitialPrice: %s\n", marketId)
```

panic rather than return

---

👌